tap_typeform/__init__.py

```diff
@@ -1,112 +1,34 @@
-                                              replication_method=replication_method,
-                                              valid_replication_keys=replication_keys)
-
-        meta = metadata.to_map(meta)
-
-        if replication_keys:
-            meta = metadata.write(meta, ('properties', replication_keys[0]), 'inclusion', 'automatic')
-
-        meta = metadata.to_list(meta)
-
-        streams.append({
-            'stream': tap_stream_id,
-            'tap_stream_id': tap_stream_id,
-            'key_properties': key_properties,
-            'schema': schema,
-            'metadata': meta,
-            'replication_method': replication_method,
-            'replication_key': replication_keys[0] if replication_keys else None
-        })
-    return Catalog.from_dict({'streams': streams})
-
-
-# this is already defined in schemas.py though w/o dependencies.  do we keep this for the sync?
-def load_schema(tap_stream_id):
-    path = "schemas/{}.json".format(tap_stream_id)
-    schema = utils.load_json(get_abs_path(path))
-    dependencies = schema.pop("tap_schema_dependencies", [])
-    refs = {}
-    for sub_stream_id in dependencies:
-        refs[sub_stream_id] = load_schema(sub_stream_id)
-    if refs:
-        singer.resolve_schema_references(schema, refs)
-    return schema
-
-
-def sync(atx):
-
-    # write schemas for selected streams\
-    for stream in atx.catalog.streams:
-        if stream.tap_stream_id in atx.selected_stream_ids:
-            schemas.load_and_write_schema(stream.tap_stream_id)
-
-    # since there is only one set of schemas for all forms, they will always be selected
-    streams.sync_forms(atx)
-
-    LOGGER.info('--------------------')
-    for stream_name, stream_count in atx.counts.items():
-        LOGGER.info('%s: %d', stream_name, stream_count)
-    LOGGER.info('--------------------')
-
-
-def _compare_forms(config_forms, api_forms):
-    return config_forms.difference(api_forms)
-
-
-def _forms_to_list(config, keyword='forms'):
-    """Splits entries into a list and strips out surrounding blank spaces"""
-    return list(map(str.strip, config.get(keyword).split(',')))
-
-
 def validate_form_ids(config):
     """Validate the form ids passed in the config"""
     client = Client(config)
+    form_stream = Forms()
 
     if not config.get('forms'):
         LOGGER.fatal("No forms were provided in config")
         raise NoFormsProvidedError
 
-    config_forms = set(_forms_to_list(config))
-    api_forms = {form.get('id') for form in client.get_forms()}
+    config_forms = _forms_to_list(config)
+    api_forms = {form.get('id') for res in form_stream.get_forms(client) for form in res}
 
-    mismatched_forms = _compare_forms(config_forms, api_forms)
+    mismatched_forms = config_forms.difference(api_forms)
 
     if len(mismatched_forms) > 0:
         LOGGER.fatal(f"FormMistmatchError: forms {mismatched_forms} not returned by API")
```

Comment:
Can you please add an form mismatch message in the FormMistmatchError exception raised below instead of defining Logger for the same?

Reply:
Added the message